cdap-api/src/main/java/co/cask/cdap/api/security/securestore/SecureStore.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.security.securestore;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Provides read access to the secure store.
+ * For write access use {@link SecureStoreManager}
+ */
+public interface SecureStore {
+
+  /**
+   * @return A map of all the elements stored in the store. The map is Name -> Description
+   */
+  Map<String, String> list() throws IOException;
```

Comment:
This is a public API, it shouldn't be optimized for easy display in some particular form.